### ISSUE TITLE
Browser and "add PostGIS layer" dialog don't list any tables in databases without PostGIS installed

### ISSUE THREAD

**Issue by andywicht** [qgis/QGIS#34966]
When trying to load a geometry-less table in QGIS 3.10.3 / 3.12.0 / master (OSGeo4W) via the Browser widget the user gets the error message "Failed to get layers" instead of the table name. In QGIS 3.4.11 it works (I don't have another version at my hands).

Trying to load via the DBManager Plugin works as expected (here just the known issue of PK detection shows).

OS: Windows 10 64 bit

**Comment by gioman** [comment]
> When trying to load a geometry-less table in QGIS 3.10.3 / 3.12.0 / master (OSGeo4W) via the Browser widget the user gets the error message "Failed to get layers" instead of the table name. In QGIS 3.4.11 it works (I don't have another version at my hands).

@andywicht cannot confirm in any of the above version, with osgeo4w on the same platform. Please add more details (and a dump of a table that does not work for you).

**Comment by andywicht** [comment]
```sql
CREATE TABLE test (
    id int4 NOT NULL GENERATED ALWAYS AS IDENTITY,
    c_name text NOT NULL,    
    CONSTRAINT pk_test PRIMARY KEY (id)
);

INSERT INTO test (c_name) VALUES('abc');
```
PostgreSQL 10.12 (Ubuntu 10.12-2.pgdg16.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609, 64-bit

![image](https://user-images.githubusercontent.com/21099522/76401898-e8272800-6382-11ea-80c4-457478b0146d.png)

It seems that this only happens when the PostGIS extension does not exist in the database.

For databases with PostGIS foreign tables are not showing and there is no error message. No idea if that is related or two separate issues.

**Comment by gioman** [comment]
@andywicht confirmed. This applies to both the QGIS browser and the "add PostGIS layer" dialog in the data source  manager, while DB Manager is ok.

After adding the PostGIS extension the tables start to show up.

It is of course an inconsistency but I can't rule out that is by design, @elpaso thoughts?

**Comment by elpaso** [comment]
Better ask @strk that worked on the provider SQL recently.

**Comment by gioman** [comment]
@strk The "add postgis layer" dialog is ok on 2.18, meaning that even without PostGIS it allows to see and add tables, so at least this part is a regression. The browser worked the same way.

**Comment by strk** [comment]
Confirmed. Logs show an error in the SQL query which assumes "geometry_columns" exists. I'm on it.
I guess older versions would still log this error, but it was JUST for the query inspecting geometry_columns, and not for the NOW-single query inspecting all tables.

**Comment by strk** [comment]
Found and fixed with #35019 (master branch) and #35018 (backport)

Basically I made the code stricter to fail on database failures. Previous versions did continue instead.
I'd rather not send erroneous queries but for now restoring the functionality would be good. Once we have tests it'll be good to avoid sending erroneous queries to DB...
